src/core/mac/mac.cpp

```diff
@@ -1677,35 +1685,29 @@ otError Mac::ProcessEnhAckSecurity(TxFrame &aTxFrame, RxFrame &aAckFrame)
 
     VerifyOrExit(txKeyId == ackKeyId, OT_NOOP);
 
+    IgnoreError(aAckFrame.GetFrameCounter(frameCounter));
+    otLogDebgMac("Rx security - Ack frame counter %u", frameCounter);
+
     IgnoreError(aAckFrame.GetSrcAddr(srcAddr));
+    IgnoreError(aTxFrame.GetDstAddr(dstAddr));
 
-    if (srcAddr.IsShort())
+    if (!srcAddr.IsNone())
     {
         neighbor = Get<NeighborTable>().FindNeighbor(srcAddr);
-
-        if (neighbor != nullptr)
-        {
-            srcAddr.SetExtended(neighbor->GetExtAddress());
-        }
     }
 
-    if (!srcAddr.IsExtended())
+    if (neighbor == nullptr && !dstAddr.IsNone())
```

Comment:
Any case response source is not None and still need to find the neighbor according to the original destination address?

Reply:
Updated as suggested